android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIListView.java

```diff
@@ -81,6 +83,15 @@ private void processProperty(String name, Object value)
 			this.listView.getRecyclerView().setScrollEnabled(isScrollable);
 		}
 
+		if (name.equals(TiC.PROPERTY_GRID_COLUMNS)) {
+			// create a simple GridLayout
+			final int gridColumns = TiConvert.toInt(value, 0);
+			if (gridColumns > 1) {
```

Comment:
What happens if we attempt to set this back to 0 after initially setting it >1?

Reply:
now it works! I've only used it "at creation" but you are right: it can be changed afterwards and I need to set it back to linear again.